### PR TITLE
Fix multi-device NaN check in KFAC training step

### PR DESCRIPTION
**Pull request by Qianruipku**
Replace direct boolean check with jnp.any() to handle loss arrays from multiple devices when detecting NaN values.

### CODE REVIEW

**Review by jsspencer — APPROVED** [review]

File: ferminet/train.py [hunk 1]
```diff
@@ -979,8 +979,9 @@ def learning_rate_schedule(t_: jnp.ndarray) -> jnp.ndarray:
       loss = loss[0]
       # per batch variance isn't informative. Use weighted mean and variance
       # instead.
-      weighted_stats = statistics.exponentialy_weighted_stats(
-          alpha=0.1, observation=loss, previous_stats=weighted_stats)
+      if not jnp.isnan(loss):
```
Comment (jsspencer) [inline ferminet/train.py:982]: Is this check necessary? I think it's ok for the weighted mean to be NaN if the loss is NaN. An alternative is to use single assignment form here, so weighted_stats is guaranteed to be a `WeightedStats` object, eg [untested]

```
if jnp.isnan(loss):
   weighted_stats = statistics.exponentially_weighted_stats(alpha=0.1, observation=weighted_stats.loss if weighed_stats is not None else 0.0, previous_stats=weighted_stats)
else:
  weighted_stats = statistics.exponentially_weighted_stats(alpha=0.1, observation=loss, previous_stats=weighted_stats)
```

**Review by jsspencer — APPROVED** [review]
